src/rich-log/MessageBuilder.hh

```diff
@@ -14,20 +14,29 @@ namespace rlog
 class MessageBuilder
 {
     // options
-private:
+public:
     void configure(prefix const& p) { _prefix = p.value; }
-    void configure(channel c) { _channel = c; }
     void configure(sep const& s) { _sep = s.value; }
     void configure(no_sep_t) { _sep = ""; }
+    void configure(err_out_t) { _use_err_stream = true; }
+    void configure(location const& loc) { _location = &loc; }
 
-    // formatted message
-public:
-    template <class... Args>
-    MessageBuilder& operator()(char const* fmt, Args&&... args)
+    void configure(info_t) { _prefix = "[info] "; }
+    void configure(warning_t)
     {
-        // TODO
-        return *this;
+        _prefix = "[warn] ";
```

Comment:
I'm not convinced that this should be implemented via prefix. in my mind, channels and prefixes are orthogonal concepts. we can keep it for now because it's not that much of an API change later on but just something to keep in mind